api/files.py

```diff
@@ -165,3 +158,45 @@ def guess_type_from_filename(filename):
     else:
         filetype = None
     return filetype
+
+
+def get_valid_file(file_info):
+    file_id = file_info.get('_id', '')
+    file_hash = file_info.get('hash', '')
+    file_uuid_path = None
+    file_hash_path = None
+
+    if file_hash:
+        file_hash_path = util.path_from_hash(file_hash)
+
+    if file_id:
+        file_uuid_path = util.path_from_uuid(file_id)
+
+    if config.support_legacy_fs:
+        if file_hash_path and config.legacy_fs.isfile(file_hash_path):
+            return file_hash_path, config.legacy_fs
+        elif file_uuid_path and config.legacy_fs.isfile(file_uuid_path):
+            return file_uuid_path, config.legacy_fs
+
+    if file_uuid_path and config.fs.isfile(file_uuid_path):
+        return file_uuid_path, config.fs
+    else:
+        raise fs.errors.ResourceNotFound('File not found: %s', file_info['name'])
+
+
+def get_signed_url(file_path, file_system, filename=None):
+    try:
+        if hasattr(file_system, 'get_signed_url'):
```

Comment:
I made a change here from checking if `config.fs` has the method `get_signed_url` to checking if `file_system` does. I believe that's what we'll want to do while supporting legacy file system in the transition state, let me know if that change is incorrect. Ryan was seeing this issue before the fix:
```
(most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/webapp2.py", line 570, in dispatch
    return method(*args, **kwargs)
  File "./api/handlers/listhandler.py", line 482, in get
    signed_url = files.get_signed_url(file_path, file_system, filename=filename)
  File "./api/files.py", line 190, in get_signed_url
    return file_system.get_signed_url(file_path, filename=filename)
AttributeError: 'OSFS' object has no attribute 'get_signed_url'
 request_id=e5c40c00-1518721379
```

Reply:
That was when attempting to access an unmigrated file when config.fs was set to a gcs file system.